File: src/bindings/csharp/test/annotation/TestModelHistory_newSetters.cs
/// 
///  @file    TestModelHistory_newSetters.cs
///  @brief   ModelHistory unit tests
///  @author  Frank Bergmann (Csharp conversion)
///  @author  Akiya Jouraku (Csharp conversion)
///  @author  Sarah Keating 
/// 
///  $Id$
///  $HeadURL$
/// 
///  This test file was converted from src/sbml/test/TestModelHistory_newSetters.c
///  with the help of conversion sciprt (ctest_converter.pl).
/// 
/// <!---------------------------------------------------------------------------
///  This file is part of libSBML.  Please visit http://sbml.org for more
///  information about SBML, and the latest version of libSBML.
/// 
///  Copyright 2005-2010 California Institute of Technology.
///  Copyright 2002-2005 California Institute of Technology and
///                      Japan Science and Technology Corporation.
///  
///  This library is free software; you can redistribute it and/or modify it
///  under the terms of the GNU Lesser General Public License as published by
///  the Free Software Foundation.  A copy of the license agreement is provided
///  in the file named "LICENSE.txt" included with this software distribution
///  and also available online as http://sbml.org/software/libsbml/license.html
/// --------------------------------------------------------------------------->*/


namespace LibSBMLCSTest {

  using libsbml;

  using  System.IO;

  public class TestModelHistory_newSetters {
    public class AssertionError : System.Exception 
    {
      public AssertionError() : base()
      {
        
      }
    }


    static void assertTrue(bool condition)
    {
      if (condition == true)
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertEquals(object a, object b)
    {
      if ( (a == null) && (b == null) )
      {
        return;
      }
      else if (a.Equals(b))
      {
        return;
      }
  
      throw new AssertionError();
    }

    static void assertNotEquals(object a, object b)
    {
      if ( (a == null) && (b == null) )
      {
        throw new AssertionError();
      }
      else if (a.Equals(b))
      {
        throw new AssertionError();
      }
    }

    static void assertEquals(bool a, bool b)
    {
      if ( a == b )
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertNotEquals(bool a, bool b)
    {
      if ( a != b )
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertEquals(int a, int b)
    {
      if ( a == b )
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertNotEquals(int a, int b)
    {
      if ( a != b )
      {
        return;
      }
      throw new AssertionError();
    }


    public void test_ModelHistory_addCreator1()
    {
      ModelHistory mh = new  ModelHistory();
      ModelCreator mc = new  ModelCreator();
      mc.setFamilyName( "Keating");
      mc.setGivenName( "Sarah");
      long i = mh.addCreator(mc);
      assertTrue( i == libsbml.LIBSBML_OPERATION_SUCCESS );
      assertTrue( mh.getNumCreators() == 1 );
      mc = null;
      mh = null;
    }

    public void test_ModelHistory_addCreator2()
    {
      ModelHistory mh = new  ModelHistory();
      ModelCreator mc = new  ModelCreator();
      mc.setGivenName( "Sarah");
      long i = mh.addCreator(mc);
      assertTrue( i == libsbml.LIBSBML_INVALID_OBJECT );
      assertTrue( mh.getNumCreators() == 0 );
      mc = null;
      mh = null;
    }

    public void test_ModelHistory_addCreator3()
    {
      ModelHistory mh = new  ModelHistory();
      ModelCreator mc = null;
      long i = mh.addCreator(mc);
      assertTrue( i == libsbml.LIBSBML_OPERATION_FAILED );
      assertTrue( mh.getNumCreators() == 0 );
      mh = null;
    }

    public void test_ModelHistory_setCreatedDate1()
    {
      ModelHistory mh = new  ModelHistory();
      assertTrue( mh != null );
      Date date = new  Date("2005-12-30T12:15:32+02:00");
      long i = mh.setCreatedDate(date);
      assertTrue( i == libsbml.LIBSBML_OPERATION_SUCCESS );
      assertTrue( mh.isSetCreatedDate() == true );
      assertTrue( date != mh.getCreatedDate() );
      string dateChar = mh.getCreatedDate().getDateAsString();
      assertTrue((  "2005-12-30T12:15:32+02:00" == dateChar ));
      i = mh.setCreatedDate(null);
      assertTrue( i == libsbml.LIBSBML_OPERATION_SUCCESS );
      assertTrue( mh.isSetCreatedDate() == false );
      date = null;
      mh = null;
    }

    public void test_ModelHistory_setCreatedDate2()
    {
      ModelHistory mh = new  ModelHistory();
      assertTrue( mh != null );
      Date date = new  Date("Jan 12");
      long i = mh.setCreatedDate(date);
      assertTrue( i == libsbml.LIBSBML_INVALID_OBJECT );
      assertTrue( mh.isSetCreatedDate() == false );
      date = null;
      mh = null;
    }

    public void test_ModelHistory_setModifiedDate1()
    {
      ModelHistory mh = new  ModelHistory();
      assertTrue( mh != null );
      Date date = new  Date("2005-12-30T12:15:32+02:00");
      long i = mh.setModifiedDate(date);
      assertTrue( i == libsbml.LIBSBML_OPERATION_SUCCESS );
      assertTrue( mh.isSetModifiedDate() == true );
      assertTrue( date != mh.getModifiedDate() );
      string dateChar = mh.getModifiedDate().getDateAsString();
      assertTrue((  "2005-12-30T12:15:32+02:00" == dateChar ));
      i = mh.setModifiedDate(null);
      assertTrue( i == libsbml.LIBSBML_OPERATION_FAILED );
      assertTrue( mh.isSetModifiedDate() == true );
      date = null;
      mh = null;
    }

    public void test_ModelHistory_setModifiedDate2()
    {
      ModelHistory mh = new  ModelHistory();
      assertTrue( mh != null );
      Date date = new  Date(200,13,76,56,89,90,0,0,0);
      long i = mh.setModifiedDate(date);
      assertTrue( i == libsbml.LIBSBML_INVALID_OBJECT );
      assertTrue( mh.isSetModifiedDate() == false );
      date = null;
      mh = null;
    }

  }
}

File: src/bindings/python/test/annotation/TestModelHistory_newSetters.py
#
# @file    TestModelHistory_newSetters.py
# @brief   ModelHistory unit tests
#
# @author  Akiya Jouraku (Python conversion)
# @author  Sarah Keating 
#
# $Id$
# $HeadURL$
#
# This test file was converted from src/sbml/test/TestModelHistory_newSetters.c
# with the help of conversion sciprt (ctest_converter.pl).
#
#<!---------------------------------------------------------------------------
# This file is part of libSBML.  Please visit http://sbml.org for more
# information about SBML, and the latest version of libSBML.
#
# Copyright 2005-2010 California Institute of Technology.
# Copyright 2002-2005 California Institute of Technology and
#                     Japan Science and Technology Corporation.
# 
# This library is free software; you can redistribute it and/or modify it
# under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation.  A copy of the license agreement is provided
# in the file named "LICENSE.txt" included with this software distribution
# and also available online as http://sbml.org/software/libsbml/license.html
#--------------------------------------------------------------------------->*/
import sys
import unittest
import libsbml

class TestModelHistory_newSetters(unittest.TestCase):


  def test_ModelHistory_addCreator1(self):
    mh = libsbml.ModelHistory()
    mc = libsbml.ModelCreator()
    mc.setFamilyName( "Keating")
    mc.setGivenName( "Sarah")
    i = mh.addCreator(mc)
    self.assert_( i == libsbml.LIBSBML_OPERATION_SUCCESS )
    self.assert_( mh.getNumCreators() == 1 )
    mc = None
    mh = None
    pass  

  def test_ModelHistory_addCreator2(self):
    mh = libsbml.ModelHistory()
    mc = libsbml.ModelCreator()
    mc.setGivenName( "Sarah")
    i = mh.addCreator(mc)
    self.assert_( i == libsbml.LIBSBML_INVALID_OBJECT )
    self.assert_( mh.getNumCreators() == 0 )
    mc = None
    mh = None
    pass  

  def test_ModelHistory_addCreator3(self):
    mh = libsbml.ModelHistory()
    mc = None
    i = mh.addCreator(mc)
    self.assert_( i == libsbml.LIBSBML_OPERATION_FAILED )
    self.assert_( mh.getNumCreators() == 0 )
    mh = None
    pass  

  def test_ModelHistory_setCreatedDate1(self):
    mh = libsbml.ModelHistory()
    self.assert_( mh != None )
    date = libsbml.Date("2005-12-30T12:15:32+02:00")
    i = mh.setCreatedDate(date)
    self.assert_( i == libsbml.LIBSBML_OPERATION_SUCCESS )
    self.assert_( mh.isSetCreatedDate() == True )
    self.assert_( date != mh.getCreatedDate() )
    dateChar = mh.getCreatedDate().getDateAsString()
    self.assert_((  "2005-12-30T12:15:32+02:00" == dateChar ))
    i = mh.setCreatedDate(None)
    self.assert_( i == libsbml.LIBSBML_OPERATION_SUCCESS )
    self.assert_( mh.isSetCreatedDate() == False )
    date = None
    mh = None
    pass  

  def test_ModelHistory_setCreatedDate2(self):
    mh = libsbml.ModelHistory()
    self.assert_( mh != None )
    date = libsbml.Date("Jan 12")
    i = mh.setCreatedDate(date)
    self.assert_( i == libsbml.LIBSBML_INVALID_OBJECT )
    self.assert_( mh.isSetCreatedDate() == False )
    date = None
    mh = None
    pass  

  def test_ModelHistory_setModifiedDate1(self):
    mh = libsbml.ModelHistory()
    self.assert_( mh != None )
    date = libsbml.Date("2005-12-30T12:15:32+02:00")
    i = mh.setModifiedDate(date)
    self.assert_( i == libsbml.LIBSBML_OPERATION_SUCCESS )
    self.assert_( mh.isSetModifiedDate() == True )
    self.assert_( date != mh.getModifiedDate() )
    dateChar = mh.getModifiedDate().getDateAsString()
    self.assert_((  "2005-12-30T12:15:32+02:00" == dateChar ))
    i = mh.setModifiedDate(None)
    self.assert_( i == libsbml.LIBSBML_OPERATION_FAILED )
    self.assert_( mh.isSetModifiedDate() == True )
    date = None
    mh = None
    pass  

  def test_ModelHistory_setModifiedDate2(self):
    mh = libsbml.ModelHistory()
    self.assert_( mh != None )
    date = libsbml.Date(200,13,76,56,89,90,0,0,0)
    i = mh.setModifiedDate(date)
    self.assert_( i == libsbml.LIBSBML_INVALID_OBJECT )
    self.assert_( mh.isSetModifiedDate() == False )
    date = None
    mh = None
    pass  

def suite():
  suite = unittest.TestSuite()
  suite.addTest(unittest.makeSuite(TestModelHistory_newSetters))

  return suite

if __name__ == "__main__":
  if unittest.TextTestRunner(verbosity=1).run(suite()).wasSuccessful() :
    sys.exit(0)
  else:
    sys.exit(1)
